### ISSUE TITLE
Profile page not working properly

### ISSUE THREAD

**Issue by catalinpit** [thepracticaldev/dev.to#7434]
**Describe the bug**

You are allowed to change the colour for some of the information from your profile page. There are cases when the profile page doesn't work properly when the colours are related.

See the pictures below or visit my profile [dev.to/catalinmpit](https://dev.to/catalinmpit).

**To Reproduce**

1. Go to [dev.to/catalinmpit](https://dev.to/catalinmpit)

**Expected behavior**

My name and other information should be visible.

**Screenshots**

![broken-profile-page](https://i.imgur.com/AJyM7tH.png)

**Aditional information**

Played around with the settings and I've done this:

![profile-page-ok](https://i.imgur.com/DLiPXFT.png)

**Question**

What do you recommend to fix the issue? If you have any recommendations, I'm happy to pick the issue.

**Comment by rhymes** [comment]
I tyink this is related to the following issues:

- [Profile Name & Social Profile Icons (GitHub/Medium) Not Visible in ten x hacker Theme](https://github.com/thepracticaldev/dev.to/issues/4811)
- [Need Contrast in Follow Option in the Dark Mode](https://github.com/thepracticaldev/dev.to/issues/3968)
- [Give contrast guidance on user profile background/text color](https://github.com/thepracticaldev/dev.to/issues/1774)
- [Disallow buttons with the same color as the background color](https://github.com/thepracticaldev/dev.to/issues/735)

I think adding a constrast indicator as suggested in https://github.com/thepracticaldev/dev.to/issues/735#issuecomment-539931733 is probably the best way to handle this.

The open question is: how to fit that with multiple themes? If I set a contrast while I'm using the light theme and then a user visits my profile using the dark theme, we'd probably be back at square one, won't we?

Personally I don't know how to handle this in a bullet proof way, maybe you have an idea @catalinpit ?

cc. @ludwiczakpawel and @nickytonline who might have suggestions

**Comment by nickytonline** [comment]
* My first thought is disallowing certain colours or colour ranges that can affect the visibility in certain themes with a friendly notification explaining why.

* A bigger idea which may be overkill is to preview colour changes in the various themes.

* One other thought is potentially disallowing changing the colours as it can cause issues as mentioned above. The last idea is not ideal,  as it's kind of fun changing the colours.

cc: @ludwiczakpawel

**Comment by catalinpit** [comment]
What do you think, @ludwiczakpawel?

**Comment by ludwiczakpawel** [comment]
> disallowing certain colours or colour ranges that can affect the visibility in certain themes with a friendly notification explaining why

that's gonna be a huge project and quite hard to handle and maintain because we have (currently) 5 themes (well, technically 3 in terms of colors) so I'd be a bit scared of this one to be honest.

> bigger idea which may be overkill is to preview colour changes in the various themes

my concerns would be the same as in previous one I think... 

> One other thought is potentially disallowing changing the colours as it can cause issues as mentioned above. The last idea is not ideal, as it's kind of fun changing the colours

Agree with you @nickytonline - even though I'm personally not a huge fan of being able to customize colors, I think it's important feature from our product perspective so killing that may not be an option for now. Eventually in future we may come up with some alternative ways for users to express their "brand and personality" rather than simply two colors but that's far far away.

> I think adding a contrast indicator as suggested in #735 (comment) is probably the best way to handle this.

@rhymes I think this may not solve the issue because it's not about contrast ratio between these two colors but about contrast ratio between one of these colors and theme color scheme. ALTHOUGH I do think this would be a neat feature anyway, just not solving the problem here :P 

### Solution?

To sum up, I think there's no ideal solution :D I think we have to rely on users - that they will, in most cases, pick colors that simply are readable. I think one potential idea is to limit elements that inherit brand colors. I'd definitely skip brand colors for name and others text in that header box on profile. Instead, I'd apply these colors to only:
- border around header box (so as it is now)
- follow button (maybe?)

And that's it. Thoughts?

(also, sorry for a bit late reply, it's been busy week)

**Comment by catalinpit** [comment]
Let's do that!

**Comment by rhymes** [comment]
This is related https://github.com/thepracticaldev/dev.to/issues/7600

**Comment by catalinpit** [comment]
This is quite annoying. Assign this to me, please.

**Comment by Amorpheuz** [comment]
Related to #7600 but mentioning it here since its being handled/assigned in this issue. Something I noticed was that under `Branding` inside settings, `Brand Color 2` is mentioned as the color for texts, so why not switch the email to the same color as that would be more concise with what is being indicated to the user.

![image](https://user-images.githubusercontent.com/22113778/81790490-1bbd3480-9523-11ea-858f-b6a5430e2fe7.png)
